Styling updates

metzlr opened the pull request:
I changed all of the CSS style sheets to SASS, replacing some of the messier CSS syntax with SASS.

I also created a new variables.scss file and began to consolidate re-used/theme related styles (e.g. colors, shadows) into variables that are used across the site.

I updated the ListItem styling some more, replacing the bootstrap styling and adding a new feature where a row is highlighted when its corresponding item is selected in the editor. Here's what that looks like right now:

![Screenshot_2020-11-13 React App](https://user-images.githubusercontent.com/54820894/99097611-b529d980-258c-11eb-8b71-b5b462d70478.png)

gmisail commented:
I changed the `background-color` property to the background color of the site, and I replaced the old color and its instances with the `dropdown-background-color` since that was its only usage. Also, I made the blue highlight a little less intense, since I thought it should be a little more subtle.

![image](https://user-images.githubusercontent.com/8661089/99103095-0944b600-25ad-11eb-82f2-77d89bdd07a1.png)

metzlr commented:
Ok so what started out as just a couple of things ended in me changing/fixing a ton of the styling across the site. Some of the things I did are:

- Item list is scrollable when items pass the bottom of the page
- Re-worked the RoomRoute grid sizing.
- Installed and setup Popper.js to handle positioning of the item dropdown menu
- Added a footer to the editor that currently displays the "1 Cell = 1 Square foot" text. Also changed how the canvas is positioned in its container.
- Removed the fixed add item button when on larger screens because it didn't look good, especially on really wide screens (its still there though once the columns become stacked)
- Added some more font weights. 
- Added some basic animations to the editor buttons and currently selected item row (I had to remove the pop out animation because it overflows the container, or would if it was able to, but because the item list hides overflow for the scrolling effect, the enlarged list item gets cutoff. Might be able to work around this in the future, but for now it isn't worth it)
- Removed the bootstrap hamburger button on the navbar that wasn't being used
- Other minor things

Try it out, see if it looks good

metzlr commented:
The left-start placement looks good, didn't notice any of the issues from before.
